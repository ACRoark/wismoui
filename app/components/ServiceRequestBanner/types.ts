export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
