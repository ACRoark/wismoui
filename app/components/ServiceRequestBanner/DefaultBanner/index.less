@import '../../../styles/themes/index.less';

@service-request-banner-cls-prefix: ~'@{dte-prefix}-default-banner';
@mobile: ~'only screen and (max-width: 365px)';
@desktop: ~'only screen and (min-width: 365px)';

.@{service-request-banner-cls-prefix} {
  background-color: @dte-gray-200;
  margin-bottom: 48px;

  > h4 {
    display: inline-block;
    font-weight: @dte-wismo-font-weight-semi-bold;
    line-height: 1.5;
    margin-block-end: 0px;
    margin-block-start: 0px;
    overflow-wrap: normal;
    padding: 20px 30px;
  }
}
