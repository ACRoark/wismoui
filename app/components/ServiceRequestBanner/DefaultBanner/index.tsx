import React, { ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import messages from 'components/ServiceRequestBanner/messages';

import { IMessage } from '../types';
import './index.less';

interface IDefaultBannerProps {
  message: IMessage;
}

const getMessage = (id: string, values: string[] | null): ReactElement => {
  if (values) {
    if (values.length > 1) {
      return (
        <FormattedMessage
          {...messages[id]}
          values={{
            date1: <DateStamp value={values[0]} />,
            date2: <DateStamp value={values[1]} />,
          }}
        />
      );
    }

    return <FormattedMessage {...messages[id]} values={{ date: <DateStamp value={values[0]} /> }} />;
  }

  return <FormattedMessage {...messages[id]} values={{ date: null }} />;
};

const DefaultBanner: React.FC<IDefaultBannerProps> = (props: IDefaultBannerProps): React.ReactElement => {
  const {
    message: { id, values },
  } = props;

  const message = getMessage(id, values);

  return (
    <div className="dte-wismo-default-banner">
      <h4>{message}</h4>
    </div>
  );
};

export default DefaultBanner;
