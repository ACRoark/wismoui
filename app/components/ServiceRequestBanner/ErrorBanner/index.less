@import '../../../styles/themes/index.less';

@service-request-error-banner-cls-prefix: ~'@{dte-prefix}-error-banner';

.@{service-request-error-banner-cls-prefix} {
  background-color: @dte-background-error;
  display: flex;
  flex-direction: row;
  margin-bottom: 48px;
  padding: 20px;

  > div > h4 {
    color: @dte-red !important;
    display: inline;
    font-weight: @dte-wismo-font-weight-semi-bold;
    line-height: 1.5;
    margin-block-end: 0;
    margin-block-start: 0;
    margin-right: 20px;
    overflow-wrap: normal;
  }

  .content {
    margin-left: 20px;

    @media @mobile {
      align-items: flex-start;
      display: flex;
      flex-direction: column;
    }
  }

  .error-icon {
    color: @dte-red;
    display: inline-block;
    margin: 50% 0;
    vertical-align: text-top;
  }

  .view-details-link {
    font-size: @dte-wismo-font-size-0;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding: unset !important;
  }
}
