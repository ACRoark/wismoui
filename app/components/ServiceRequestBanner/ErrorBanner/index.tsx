import React from 'react';
import { FormattedMessage } from 'react-intl';

import ExclamationPoint from 'images/ExclamationPoint';

import messages from '../messages';
import { IMessage } from '../types';
import './index.less';

interface IErrorBannerProps {
  message: IMessage;
}

const ErrorBanner: React.FC<IErrorBannerProps> = (props: IErrorBannerProps): React.ReactElement => {
  const {
    message: { id },
  } = props;

  return (
    <div className="dte-wismo-error-banner">
      <div>
        <ExclamationPoint className="error-icon" title="error" />
      </div>
      <div className="content">
        <h4>
          <FormattedMessage {...messages[id]} values={{ date: null }} />
        </h4>
      </div>
    </div>
  );
};

export default ErrorBanner;
