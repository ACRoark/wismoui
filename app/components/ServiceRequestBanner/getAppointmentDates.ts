import { IAppointmentList } from 'types';

const getAppointmentDates = (appointmentList: IAppointmentList[]): string[] => {
  const dates: string[] = [];

  appointmentList.forEach((appointment: IAppointmentList): void => {
    if (!appointment.completed) {
      dates.push(appointment.date);
    }
  });

  return dates;
};

export default getAppointmentDates;
