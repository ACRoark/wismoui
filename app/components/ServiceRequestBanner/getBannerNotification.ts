import * as _ from 'lodash';

import createAppointmentList from 'components/Stepper/createAppointmentList';
import { IAppointmentList, IOrderRequest, IServiceOrderEvent } from 'types';
import filterVerificationEvents from 'utils/filterVerificationEvents';

import getAppointmentDates from './getAppointmentDates';

import { IBannerNotification } from './types';

const getBannerNotification = (serviceRequest: IOrderRequest): IBannerNotification => {
  const { verificationEvents } = serviceRequest;

  const appointmentList: IAppointmentList[] = createAppointmentList(serviceRequest);
  const dates = getAppointmentDates(appointmentList);
  const hasMultipleDates = dates.length > 1;

  const verificationCodes = ['AD', 'AN', 'CL', 'DR', 'ES', 'FV', 'FW', 'LD', 'PR', 'RE', 'RF', 'SC', 'WP'];
  const filteredVerificationEvents = filterVerificationEvents(verificationEvents, verificationCodes);

  if (filteredVerificationEvents.length) {
    const id = _.last(filteredVerificationEvents[0].verificationEventStatusUpdates).verificationCode;

    return {
      hasError: true,
      message: {
        id,
        values: null,
      },
    };
  }

  const allAppointments: IServiceOrderEvent[] = appointmentList.reduce(
    (appointments: IServiceOrderEvent[], appointmentsByDate: IAppointmentList): IServiceOrderEvent[] => {
      if (!appointmentsByDate.completed) {
        return appointments.concat(appointmentsByDate.appointments);
      }
      return appointments;
    },
    [],
  );

  const delayedErrorAppointment = allAppointments.find((appointment: IServiceOrderEvent): boolean =>
    ['DELAYED'].includes(`${appointment.latestStatus.serviceOrderEventStatus}`),
  );

  const errorAppointment = allAppointments.find((appointment: IServiceOrderEvent): boolean =>
    ['CALL_DTE', 'CANCELED_CGI', 'CANCELED_INC', 'PARTIAL_COMPLETION', 'RESCHEDULED'].includes(
      `${appointment.latestStatus.serviceOrderEventStatus}`,
    ),
  );

  if (delayedErrorAppointment || errorAppointment) {
    const id = `${
      delayedErrorAppointment?.latestStatus.serviceOrderEventStatus ||
      errorAppointment?.latestStatus.serviceOrderEventStatus
    }`;

    return {
      hasError: true,
      message: {
        id,
        values: null,
      },
    };
  }

  return {
    hasError: false,
    message: {
      id: hasMultipleDates ? `${serviceRequest.orderRequestType}_multiple_dates` : serviceRequest.orderRequestType,
      values: dates.length === 0 ? null : dates,
    },
  };
};

export default getBannerNotification;
