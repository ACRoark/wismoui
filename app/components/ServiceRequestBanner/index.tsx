import React from 'react';

import { IOrderRequest } from 'types';

import DefaultBanner from './DefaultBanner';
import ErrorBanner from './ErrorBanner';
import getBannerNotification from './getBannerNotification';
import { IBannerNotification } from './types';

interface IServiceRequestBannerProps {
  serviceRequest: IOrderRequest;
}

const ServiceRequestBanner: React.FC<IServiceRequestBannerProps> = (
  props: IServiceRequestBannerProps,
): React.ReactElement => {
  const { serviceRequest } = props;

  const bannerNotification: IBannerNotification = getBannerNotification(serviceRequest);

  if (bannerNotification.hasError) {
    return <ErrorBanner message={bannerNotification.message} />;
  }

  return <DefaultBanner message={bannerNotification.message} />;
};

export default ServiceRequestBanner;
