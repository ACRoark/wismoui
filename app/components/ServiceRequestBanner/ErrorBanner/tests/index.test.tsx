import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import createFakeOrder from 'components/createFakeOrder';
import getBannerNotification from 'components/ServiceRequestBanner/getBannerNotification';
import { IMessage } from 'components/ServiceRequestBanner/types';
import useConfig from 'hooks/useConfig';
import useModal from 'hooks/useModal';
import {createMockStore, createSnapshotWithIntl, renderWithIntl} from 'testing/utils';

import ErrorBanner from '../index';

jest.mock('hooks/useConfig');
jest.mock('hooks/useModal');

const canceledStartService = createFakeOrder('MI12345677');
const fakeStartService = createFakeOrder('MI89012345');
const fakeStopService = createFakeOrder('MO89012345');
const fakeServiceDelayed = createFakeOrder('MO34567890');
const fakeServiceCallDTE = createFakeOrder('MO45678901');
const fakeServiceRescheduled = createFakeOrder('MO56789012');
const fakeServicePartialCompletion = createFakeOrder('MO67890123');
const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockModal = useModal as jest.MockedFunction<typeof useModal>;
const store = createMockStore();

const expectComponentToMatchSnapshot = (language: string, message: IMessage): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const tree = createSnapshotWithIntl(
    <Provider store={store}>
      <ErrorBanner message={message} />
    </Provider>, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrors = (language: string, message: IMessage): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(
    <Provider store={store}>
      <ErrorBanner message={message} />
    </Provider>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<ErrorBanner />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered with a Start Service request and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeStartService.orderRequests[0]);

      shouldNotLogErrors(language, message);
    });

    it(`should not log errors in console when rendered with a Stop Service request and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeStopService.orderRequests[0]);

      shouldNotLogErrors(language, message);
    });

    it(`should render correctly when given a Start Service request with an unresolved verification step and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeStartService.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given a Stop Service request with an unresolved verification step and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeStopService.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given the Service request with a delayed status and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeServiceDelayed.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given a Start Service request with a 'CANCELED' status and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(canceledStartService.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given the Service request with a call dte status and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeServiceCallDTE.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given the Service request with a rescheduled status and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeServiceRescheduled.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });

    it(`should render correctly when given the Service request with a partial completion status and the current language is ${language}`, (): void => {
      const { message } = getBannerNotification(fakeServicePartialCompletion.orderRequests[0]);

      expectComponentToMatchSnapshot(language, message);
    });
  });
});
