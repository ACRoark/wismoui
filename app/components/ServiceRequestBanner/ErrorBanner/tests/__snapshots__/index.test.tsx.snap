// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBanner /> should render correctly when given a Start Service request with a 'CANCELED' status and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Order Canceled
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given a Start Service request with a 'CANCELED' status and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Orden Cancelada
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given a Start Service request with an unresolved verification step and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Verification Pending
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given a Start Service request with an unresolved verification step and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Verificación Pendiente
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given a Stop Service request with an unresolved verification step and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Verification Incomplete
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given a Stop Service request with an unresolved verification step and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Verificación Incompleta
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a call dte status and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Contact Us
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a call dte status and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Contacta con nosotras
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a delayed status and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Order Delayed
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a delayed status and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Orden demorada
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a partial completion status and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Service Appointment
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a partial completion status and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Cita de servicio
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a rescheduled status and the current language is en 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Rescheduled
    </h4>
  </div>
</div>
`;

exports[`<ErrorBanner /> should render correctly when given the Service request with a rescheduled status and the current language is es 1`] = `
<div
  className="dte-wismo-error-banner"
>
  <div>
    <svg
      className="error-icon"
      fill="none"
      height="23"
      viewBox="0 0 6 23"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        error
      </title>
      <path
        d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <h4>
      Reprogramado
    </h4>
  </div>
</div>
`;
